dorado/lagrangian_walker.py

```diff
@@ -39,56 +39,187 @@ def random_pick_seed(choices, probs=None):
+    weight_sfc = (tile_domain_array(Particles.stage) \
+                  - big_sliding_window(Particles.stage))
+    weight_sfc /= tile_local_array(Particles.distances, L, W)
+    weight_sfc[weight_sfc <= 0] = 0
+    clear_borders(weight_sfc)
+    
+    # calculate inertial component weights
+    weight_int = (tile_domain_array(Particles.qx)*tile_local_array(Particles.jvec, L, W)) \
+                 + (tile_domain_array(Particles.qy)*tile_local_array(Particles.ivec, L, W))
+    weight_int /= tile_local_array(Particles.distances, L, W)
+    weight_int[weight_int <= 0] = 0
+    clear_borders(weight_int)
+    
     # get depth and cell types for neighboring cells
-    depth_ind = Particles.depth[ind[0]-1:ind[0]+2, ind[1]-1:ind[1]+2]
-    ct_ind = Particles.cell_type[ind[0]-1:ind[0]+2, ind[1]-1:ind[1]+2]
+    depth_ind = big_sliding_window(Particles.depth)
+    ct_ind = big_sliding_window(Particles.cell_type)
 
     # set weights for cells that are too shallow, or invalid 0
     weight_sfc[(depth_ind <= Particles.dry_depth) | (ct_ind == 2)] = 0
     weight_int[(depth_ind <= Particles.dry_depth) | (ct_ind == 2)] = 0
-
+    
     # if sum of weights is above 0 normalize by sum of weights
-    if nansum(weight_sfc) > 0:
-        weight_sfc = weight_sfc / nansum(weight_sfc)
-
-    # if sum of weight is above 0 normalize by sum of weights
-    if nansum(weight_int) > 0:
-        weight_int = weight_int / nansum(weight_int)
+    norm_sfc = np.nansum(weight_sfc, axis=2)
+    idx_sfc = tile_domain_array((norm_sfc > 0))
+    weight_sfc[idx_sfc] /= tile_domain_array(norm_sfc)[idx_sfc]
+    
+    norm_int = np.nansum(weight_int, axis=2)
+    idx_int = tile_domain_array((norm_int > 0))
+    weight_int[idx_int] /= tile_domain_array(norm_int)[idx_int]
 
     # define actual weight by using gamma, and weight components
     weight = Particles.gamma * weight_sfc + \
-        (1 - Particles.gamma) * weight_int
-
+             (1 - Particles.gamma) * weight_int
+    
     # modify the weight by the depth and theta weighting parameter
     weight = depth_ind ** Particles.theta * weight
-
+    
     # if the depth is below the minimum depth then location is not
     # considered therefore set the associated weight to nan
-    weight[(depth_ind <= Particles.dry_depth) | (ct_ind == 2)] \
-        = np.nan
+    weight[(depth_ind <= Particles.dry_depth) | (ct_ind == 2)] = 0
 
     # if it's a dead end with only nans and 0's, choose deepest cell
-    if nansum(weight) <= 0:
-        weight = np.zeros_like(weight)
-        weight[depth_ind == np.max(depth_ind)] = 1.0
+    zero_weights = tile_domain_array((np.nansum(weight, axis=2) <= 0))
+    deepest_cells = (depth_ind == tile_domain_array(np.max(depth_ind, axis=2)))
+    choose_deep_cells = (zero_weights & deepest_cells)
+    weight[choose_deep_cells] = 1.0
+
+    # Final checks, eliminate invalid choices
+    clear_borders(weight)
+    # weight[:,:,4] = 0. # Should particles be allowed to stand still?
```

Comment:
I think we can delete this line and opt for an issue if you want us to try to more delicately handle "standing still"